index.js

```diff
@@ -37,10 +34,31 @@ var anymatch = function(criteria, value, returnIndex, startIndex, endIndex) {
     default:
       result = false;
     }
-    if (result) { matchIndex = index + startIndex; }
+    if (result) {
+      matchIndex = index + startIndex;
+    }
     return result;
   }
-  var matched = criteria.slice(startIndex, endIndex).some(testCriteria);
+  var crit = criteria;
+  var negGlobs = crit.filter(function(criterion, index) {
```

Comment:
you could loop once with a `while`or `for` loop instead of twice with the `filter` and `map` functions. or use a `reduce`. @paulmillr already mentioned the other thing I noticed. looks good